### DISCUSSION TITLE
scvelo projections after velocity kernel and compute transition matrix look different

### DISCUSSION THREAD

**Discussion by josegarciamanteiga** [theislab/cellrank#1215] [Q&A]
Hi,
I have followed the vignette and used scvelo to calculate velocities using mode = "stochastic" and the projection looks great. Then I go with cellrank 
```
vk = cr.kernels.VelocityKernel(adata)
vk.compute_transition_matrix()
vk.plot_projection()
```

And the projections obtained on the 'vk' kernel object are different and 'worse' than the ones obtained with scvelo:

```
scv.tl.velocity_graph(adata)
scv.pl.velocity_embedding_stream(adata, basis='umap')
```

Is it ok? Why is that? I guess the scvelo transitions are a bit differently calculated compared to cell rank, but since the stream of scvelo looks great, I'd like to go forward with all the cellrank states, drivers and downstream analyses but keeping the stream I got with scvelo. Am I mistaking something? 
Thanks for the help
Jose

**Comment by WeilerP** [comment]
Hi @josegarciamanteiga, velocity projections should never form the basis of a consistent analysis. Instead, you should rely on estimated terminal states with one of our kernels, for example. Velocity projections are easily confounded and change based on different input parameters such as the genes considered, resolution parameters, etc. I presume that's exactly what's happening here. If you are interested in how exactly we compute the projects compared to scvelo, I invite you to compare the two respective code bases.
